server/src/configs/tcp.rs

```diff
@@ -13,3 +15,13 @@ pub struct TcpTlsConfig {
     pub certificate: String,
     pub password: String,
 }
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
```

Comment:
Maybe this config could look like this (no `Option<T>` types, as AFAIR there's some issue with overriding these values from config files).

```rust
#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct TcpSocketConfig {
    pub enabled: bool, // This can be named differently, just a flag saying if the default values should be overridden by a custom config, should have a default value of `false` to enable backward compatibility.
    pub ipv6: bool,
    pub keepalive: bool,
    pub nodelay: bool,
    pub recv_buffer_size: IggyByteSize,
    pub send_buffer_size: IggyByteSize,
    pub linger: IggyDuration,
}
```